services/skipchain/skipchain.go

```diff
@@ -0,0 +1,321 @@
+package skipchain
+
+import (
+	"crypto/rand"
+	"errors"
+
+	"fmt"
+
+	"bytes"
+
+	"sync"
+
+	"github.com/dedis/cothority/lib/cosi"
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/dedis/cothority/lib/network"
+	"github.com/dedis/cothority/lib/sda"
+)
+
+// ServiceName can be used to refer to the name of this service
+const ServiceName = "Skipchain"
+
+func init() {
+	sda.RegisterNewService(ServiceName, newSkipchainService)
+	skipchainSID = sda.ServiceFactory.ServiceID(ServiceName)
+}
+
+var skipchainSID sda.ServiceID
+
+// Service handles adding new SkipBlocks
+type Service struct {
+	*sda.ServiceProcessor
+	// SkipBlocks points from SkipBlockID to SkipBlock but SkipBlockID is not a valid
+	// key-type for maps, so we need to cast it to string
+	SkipBlocks map[string]*SkipBlock
```

Comment:
I didn't want to set the HashID as an `array of bytes` because if we ever change the hashing-method, we'd have to adjust the HashID-definition. Which might be OK. But still thinking (hoping) that one day we'll have more than one encryption-suite going at the same time...